### DISCUSSION TITLE
Can the operator destination_account_is be used with OR?

### DISCUSSION THREAD

**Discussion by weilah** [firefly-iii/firefly-iii#7926] [Firefly III] [answered]
I'm trying to setup a rule to match different destination account names. In summary something like:
if destination account is A or B or C then Set category to XYZ.

I tried with "Stop processing other triggers" for each trigger but it doesn't seem to work. I also tried "Strict mode" ON|OFF but didn't work either. 
I then moved to the Search feature to see if I can then create a rule out of it. Adding in the search input something like `destination_account_is:A OR destination_account_is:B OR destination_account_is:C` was also not giving me what I wanted.
I saw in the docs that `source_account_id:1,2,3` can be used. I tried to run it with `destination_account_id:1,2,3` but only the first id was getting filtered. 

Finally I tried `account_id:1,2,3` and looks like this gets me what I was looking for. 

Still I'm left with a few question marks:
Can a rule be created to match different destination account names?
Should `destination_account_id:1,2,3` be working like `account_id:1,2,3`? Maybe even `destination_account_is:A,B,C` should work like that?

Thanks!

Edit: I'm on Version [6.0.18]

**Comment by JC5** [comment]
I'm surprised, I'll test this!

**Comment by JC5** [comment, marked as answer]
The following rule works for me. Is this what you had in mind?

<img width="1458" alt="Screenshot 2023-09-03 at 17 23 35" src="https://github.com/firefly-iii/firefly-iii/assets/5889984/7fbdf24b-f711-4cbb-812c-af83317f474d">

Incidentally, this also works:

<img width="1463" alt="Screenshot 2023-09-03 at 17 26 29" src="https://github.com/firefly-iii/firefly-iii/assets/5889984/011fefe7-298b-4a5c-a207-88aafae9a4e5">

Right now, I don't believe "source_account_id" and "destination_account_id" accept a number of ID's, but that doesn't really matter since the ID also determines the account type so you know where to expect it (source or destination).

**Comment by weilah** [comment]
hey James @JC5, thanks for checking. You are right, this works. In my original post I mixed the _Rules_ feature with the _Search_ functionality, my bad. 

I read the documentation carefully and realised that what I'm trying to achieve might not be possible:
"Rules like "the budget must be Groceries and the shop must be A or B" combine different operators ("and" and "or"). This will not work."

My tests were not working because I was combining multiple "Destination account is.." triggers with "Has no category". 
What I had in mind was: if destination account is A or B or C then Set category to XYZ, if category is not set. Which then turns into a "Has no category" AND (destination account is A OR B OR C....).

So again, sorry, because the documentation states already that this is not possible. I think it would come in very handy, being able to define a category for specific expense accounts if the category is not specified when creating the transaction. But I will be able to work around it.

I'm enjoying Firefly iii very much so far!

**Reply by JC5** [reply]
Ahhh check, yes in the search it's always AND. I hope to make an OR search engine in the near future but it's pretty complex, so don't hold your breath 😉 

Thanks for the compliments!
